blockindex/sgd_indexer.go

```diff
@@ -285,9 +285,12 @@ func (sgd *sgdRegistry) StartHeight() uint64 {
 
 // PutBlock puts a block into SGDIndexer
 func (sgd *sgdRegistry) PutBlock(ctx context.Context, blk *block.Block) error {
-	if blk.Height() < sgd.startHeight {
+	if ignore, err := sgd.validateBlock(blk); err != nil {
+		return err
+	} else if ignore {
```

Comment:
this is not very clean, i think just make it same as `contractstaking.Indexer.PutBlock()` which you fixed in #3924